github-auth-provider/main.go

```diff
@@ -16,37 +16,29 @@ import (
 	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/validation"
 	"github.com/obot-platform/tools/auth-providers-common/pkg/env"
 	"github.com/obot-platform/tools/auth-providers-common/pkg/state"
+	"github.com/obot-platform/tools/github-auth-provider/pkg/config"
 	"github.com/obot-platform/tools/github-auth-provider/pkg/profile"
 	"github.com/sahilm/fuzzy"
 )
 
-type Options struct {
-	ClientID                 string  `env:"OBOT_GITHUB_AUTH_PROVIDER_CLIENT_ID"`
-	ClientSecret             string  `env:"OBOT_GITHUB_AUTH_PROVIDER_CLIENT_SECRET"`
-	ObotServerURL            string  `env:"OBOT_SERVER_URL"`
-	PostgresConnectionDSN    string  `env:"OBOT_AUTH_PROVIDER_POSTGRES_CONNECTION_DSN" optional:"true"`
-	AuthCookieSecret         string  `usage:"Secret used to encrypt cookie" env:"OBOT_AUTH_PROVIDER_COOKIE_SECRET"`
-	AuthEmailDomains         string  `usage:"Email domains allowed for authentication" default:"*" env:"OBOT_AUTH_PROVIDER_EMAIL_DOMAINS"`
-	AuthTokenRefreshDuration string  `usage:"Duration to refresh auth token after" optional:"true" default:"1h" env:"OBOT_AUTH_PROVIDER_TOKEN_REFRESH_DURATION"`
-	GitHubOrg                *string `usage:"restrict logins to members of this GitHub organization" optional:"true" env:"OBOT_GITHUB_AUTH_PROVIDER_ORG"`
-	GitHubAllowUsers         *string `usage:"users allowed to log in, even if they do not belong to the specified org and team or collaborators" optional:"true" env:"OBOT_GITHUB_AUTH_PROVIDER_ALLOW_USERS"`
-}
-
 func main() {
-	var opts Options
-	if err := env.LoadEnvForStruct(&opts); err != nil {
-		fmt.Printf("ERROR: github-auth-provider: failed to load options: %v\n", err)
-		os.Exit(1)
-	}
+	opts, err := config.LoadEnv()
+	if len(os.Args) > 1 && os.Args[1] == "validate" {
+		if err != nil {
+			var validationErr env.ValidationError
+			if errors.As(err, &validationErr) {
```

Comment:
Actually, when validation fails, you should do this and return os.exit(0) https://github.com/obot-platform/enterprise-tools/pull/64/files#diff-01e49a110328ad7a46eea37ea6af13367c4de003c836988ef175f66f9750839aR27

This is so that the backend will not interpret it as an error and just read from output(json object) to decide if it is an error.